### ISSUE TITLE
[bug] weird knee deform kkbp armature and rigify problems

### ISSUE THREAD

**Issue by multitrack-collector** [FlailingFog/KK-Blender-Porter-Pack#574]
Hi I'm using **Blender 3.6** and **mmd tools version 2.8.1,** I also use **KKBP 3.5** because with newer versions, some textures don't import properly.

Anyways, my problem is that when I import a character into Blender, the kkbp armature causes **the character's knees to deform incorrectly by shrinking more than they should**. [Here's a link to a video that shows what's happening](https://mega.nz/file/2ZUgHa4J#lPxGQ0ZsqYfLsDalvvQr4B56Qza2DWk8VZq_nxhRAEM) I believe I saw a similar issue with newer versions as well.

Though the rigify armature doesn't have this issue, moving the ik hand or foot bone results in the link being over stretched. [Here's a video that demonstrates this.](https://mega.nz/file/uREyBSrb#POMZC9Ebm05fRgdtdNlzt2rwUWua4UjTnc-DyauIPTg) 

Is there a way to fix this?

Thanks!

Here's the card I used in all videos.

![KoikatsuSun_F_20240617085756423_Yuuki Sakura](https://github.com/FlailingFog/KK-Blender-Porter-Pack/assets/166049022/e35baa0d-93cb-4d21-8af7-b32d3bed8370)

**Comment by MoriMorinya** [comment]
the rigify stretch thing is a feature, you can disable it by selecting the hand and leg bones (separately and for both sides), going to the left tabs up to item and dragging "Ik stretch" to zero. 
![nnb4cnhvxcmc1](https://github.com/user-attachments/assets/3847ffa1-473e-4abd-a92e-a173cb33c518)

using the rigify rig is perfered over the kkbp armture. less issues that way.

**Comment by multitrack-collector** [comment]
Okay Thx

**Comment by FlailingFog** [comment]
It's likely something to do with one of the driver bones on layer 3 of the KKBP armature.
I think this was fixed at some point between KKBP 6.5 and 6.6 because I'm not seeing this issue. I am actually using KKBP 7.0 to test this, but I don't believe I touched the knee drivers between 6.6 and 7.0 so it should the same thing.

Maybe try deleting the drivers on the knee deform bone or some other bone on armature layer 3 to see if that fixes it?

![image](https://github.com/user-attachments/assets/359e4c9b-fb3f-4f94-aef2-e87703f75e12)
